frontend/lib/data/repositories/database_repository_impl.dart

```diff
@@ -11,27 +11,36 @@ import 'package:note_keeper/domain/repositories/repositories.dart';
 
 final _databaseRepositoryProvider = Provider<DatabaseRepositoryImpl>((ref) =>
     DatabaseRepositoryImpl(
-        ref.read(BackendDependency.database), ref.watch(AppState.auth)));
+        ref.read(BackendDependency.database), ref.watch(AppState.auth), ref));
 
 class DatabaseRepositoryImpl extends DatabaseRepository
     with RepositoryExceptionMixin {
-  DatabaseRepositoryImpl(this._database, this._authState);
+  /// Private constructor
+
+  //below is the working constructor. but needs to have user Id as field
+  DatabaseRepositoryImpl(this._database, this._authState, this.ref);
+  String get userId => _authState.user!.$id;
 
   static Provider<DatabaseRepositoryImpl> get provider =>
       _databaseRepositoryProvider;
-
-  final Database _database;
+  final Ref ref;
```

Comment:
@mandolf0  ref is never used. no need to create Ref

Reply:
It is needed in lines 12-14 to  later define the provider => in line 24;